### ISSUE TITLE
Bug: custom indent doesn't always work

### ISSUE THREAD

**Issue by EgonBolton** [facebook/lexical#4024]
Using this: https://github.com/facebook/lexical/pull/3802
I am noticing that there are several occasions where the custom indent is not applied, and the default 20px appears. If I tab on that node, it automatically adjust to the corresponding indent.

Lexical version: 0.8.1

## Steps To Reproduce

1. Use my code-sandbox, where indent=40px and also I extend paragraphNode so that each inserted paragraph has the same indentation as the previous one.
2. Press enter on an indented paragraph. The new paragraph will have the same __indent but will be calculated over the default 20px.

Link to code example: https://codesandbox.io/s/angry-rosalind-969j13?file=/src/Editor.js

https://www.loom.com/share/998a64bb54d24b83bd50beb9ea9ee58d
I just realized that Loom didn't record devtools... but it's still understandable

## The current behavior
Custom Indent is not displayed

## The expected behavior
Custom Indent should be displayed

**Comment by thegreatercurve** [comment]
@EgonBolton Thanks for the detailed bug report!

Tbh, I'm not sure about the underlying root cause, but we might have to put a warning about trying to override `insertAfter` on the `ParagraphNode` as it's on the very base class of `LexicalNode` and its called a lot all over core, not just on pressing enter. It's possible you're getting an old cached version of the node without the indent classes applied. 

Either way, a better solution could be to add a `KEY_ENTER_COMMAND` command and then in your callback get the previous sibling of the current selection, and then get the indent from that and apply it to the current node.

**Comment by EgonBolton** [comment]
>  It's possible you're getting an old cached version of the node without the indent classes applied.

No, it is that when inserting a paragraph in enter the indent is not preserved.
You can see this by pressing enter on an indented paragraph in the playground. The new paragraph is not indented.
This is so because the `insertNewAfter` of paragraphNode is not setting indent.

About not overriding `insertAfter`, I understand what you mean, but I have a use case that is necessary.
